### PR TITLE
Feature/custom output extract faces

### PR DESCRIPTION
**Pull request by ProgramadorArtificial**
### What has been done

Changed extract_faces parameters to be possible to return face image (croped) as rgb, bgr or grayscale and flag for normalized. By default, all parameters will return the same as before (RGB and normalized).

## How to test
Change parameters as desired:
```
faces = DeepFace.extract_faces(
    img_path = 'image.jpg',
    normalize_face=True,  # True or False
    color_face='bgr'  # rgb, bgr or gray
)
```
After extracting face can save to see the result:
```
Image.fromarray((faces[0]['face']*255).astype(np.uint8)).save('img.png')  # normalize_face == True
Image.fromarray((faces[0]['face']).astype(np.uint8)).save('img.png')  # normalize_face == False
```

### CODE REVIEW

**Comment by ProgramadorArtificial** [comment]
image
I recreated the PR... answering here :)
Ok. Can I add in documentation the "grayscale" parameter is deprecated? (keeping it working)

File: deepface/modules/verification.py [hunk 1]
```diff
@@ -239,7 +239,8 @@ def __extract_faces_and_embeddings(
     img_objs = detection.extract_faces(
         img_path=img_path,
         detector_backend=detector_backend,
-        grayscale=False,
+        color_face='rgb',
+        normalize_face=True,
```
Comment (serengil) [inline deepface/modules/verification.py:243]: default values of color face and normalize face are rgb and True already, IMO we should not set them where it is being called if we are setting default values.

this is available also in representation, recognition,

Comment (ProgramadorArtificial) [reply deepface/modules/verification.py]: Removed them. I added it because grayscale was there.

File: deepface/modules/representation.py [hunk 2]
```diff
@@ -71,7 +71,6 @@ def represent(
         img_objs = detection.extract_faces(
             img_path=img_path,
             detector_backend=detector_backend,
-            grayscale=False,
```
Comment (serengil) [inline deepface/modules/representation.py:74]: restore this please

File: deepface/modules/recognition.py [hunk 3]
```diff
@@ -364,7 +363,6 @@ def __find_bulk_embeddings(
             img_objs = detection.extract_faces(
                 img_path=employee,
                 detector_backend=detector_backend,
-                grayscale=False,
```
Comment (serengil) [inline deepface/modules/recognition.py:367]: restore this please

File: deepface/modules/recognition.py [hunk 4]
```diff
@@ -240,7 +240,6 @@ def find(
     source_objs = detection.extract_faces(
         img_path=img_path,
         detector_backend=detector_backend,
-        grayscale=False,
```
Comment (serengil) [inline deepface/modules/recognition.py:243]: restore this please

File: deepface/DeepFace.py [hunk 5]
```diff
@@ -586,7 +596,8 @@ def detectFace(
         detector_backend=detector_backend,
         enforce_detection=enforce_detection,
         align=align,
-        grayscale=False,
```
Comment (serengil) [inline deepface/DeepFace.py:589]: restore this please

File: deepface/modules/demography.py [hunk 6]
```diff
@@ -123,7 +123,6 @@ def analyze(
     img_objs = detection.extract_faces(
         img_path=img_path,
         detector_backend=detector_backend,
-        grayscale=False,
```
Comment (serengil) [inline deepface/modules/demography.py:126]: restore this please

**Pull request merged**